analytics.go

```diff
@@ -100,6 +108,10 @@ func NewWithConfig(writeKey string, dataPlaneUrl string, config Config) (cli Cli
 	return
 }
 
+func (c *client) SetGZIP(gzip bool) {
+	GzipSupport = gzip
```

Comment:
Setting the global `GzipSupport` from a struct method is not ideal. The Consumer is expected to set the gzip support at the `client` / instance level, not the global one. 

I would add GzipSupport as a filed in client struct, set it to the `defaultGzipSupport` (a const global bool)

Reply:
Depending on your concurrency model, it might not be safe to update this boolean this way. If you read it in a separate go routine while you writing it, the race detector is going to complain. 

Maybe we can move this to config, and remove the method all together.

Reply:
there is already an `analytics.Config` struct, we shouldn't deviate from using it for all configuration aspects of this library.

Reply:
Do get a better understanding: Was the idea to keep `analytics.Config` the same as segment's so we can be a drop-in replacement? 

@atzoum has a valid point.

Reply:
@lvrach Segment does have support/API/config over gzip support, in fact, they don't support gzip in their Go SDK. We came up with the idea to expose an API for this & keep the parity among SDKs.

Reply:
Btw, I also do agree with @atzoum. This decision was made during my leave. And other SDKs have been implemented this way. So, we can't change the scope now.

Reply:
@atzoum @lvrach gzip has been moved to config. Can you check?